src/index/refine_flat.rs

```diff
@@ -71,7 +71,7 @@ impl<BI> NativeIndex for RefineFlatIndexImpl<BI> {
     }
 }
 
-impl<BI> FromInnerPtr for RefineFlatIndexImpl<BI> {
+impl<IndexImpl> FromInnerPtr for RefineFlatIndexImpl<IndexImpl> {
```

Comment:
Should be this instead, otherwise it's still a type parameter shadowing the struct `IndexImpl`.
```suggestion
impl FromInnerPtr for RefineFlatIndexImpl<IndexImpl> {
```